pypeerassets/protocol.py

```diff
@@ -267,64 +269,65 @@ def validate_card_issue_modes(issue_mode: int, cards: list) -> list:
     for i in [1 << x for x in range(len(IssueMode))]:
         if bool(i & issue_mode):
             print('Applying {0} parser.'.format(IssueMode(i).name))
-            cards = parsers[IssueMode(i).name](cards)
+            parser_fn = cast(Callable[[list], Optional[list]], parsers[IssueMode(i).name])
+            return parser_fn(cards)
 
-    return cards
+    return []
```

Comment:
This should actually return some cards :)

Reply:
oh, ha, yeah... that would probably be right :+1:

Reply:
i suppose it shouldn't return from the loop either, but just keep running `cards` through parsers functions

Reply:
Hm not sure I get you, can you show by example?

Reply:
Oh, i have some tests around this function in this next iteration... that might help make it all clearer :)